src/enclave-agent/src/services/images.rs

```diff
@@ -111,6 +99,29 @@ impl ImageService {
 
         Ok(image.to_owned())
     }
+
+    fn parse_container_id(&self, req: &image::PullImageRequest) -> Result<String> {
+        let mut cid = req.get_container_id().to_string() ;
```

Comment:
is the shim sending invalid container id?

Reply:
The shim in this repo will format the container id in [here](https://github.com/confidential-containers/enclave-cc/blob/c9ab4f08cfcd7afdb66c93956150e71de6225446/src/shim/runtime/v2/rune/v2/utils.go#L58).  The function `parse_container_id` makes the last check.

Reply:
I'm confused what is the error this PR fixes. If the shim sends valid `cid`s I don't see how this changes the flow. It also makes me think could we just error on all invalid (and empty) `cid`s

Reply:
According to the code, the original design mimics the kata on CCv0. The kata shim sends the pulling request to kata agent [here](https://github.com/confidential-containers/kata-containers/blob/cfa3e1e9334dedeb3e28440e6b3c3d268fc2ea14/src/runtime/pkg/containerd-shim-v2/image.go#L72), which has no container id but the image name. Therefore, the kata agent need to create and format the container id [here](https://github.com/confidential-containers/kata-containers/blob/cfa3e1e9334dedeb3e28440e6b3c3d268fc2ea14/src/agent/src/image_rpc.rs#L191) if the image name exists and the container id is empty. Otherwise the kata agent validates the container id if the the container id is provided. I find the problem because I send an invalid `cid` from a dev tool. The dev tool need to be fixed.

If we want  to error on all invalid (and empty) `cid`s, we need to enforce the client provides an non-empty `cid`. 

If we follow the kata design,  the PR can be closed.  I think it is a better choice because the it make the agent more flexible.

Reply:
@HaokunX-intel thanks, great feedback. 

> If we want to error on all invalid (and empty) cids, we need to enforce the client provides an non-empty cid.
> 
> If we follow the kata design, the PR can be closed. I think it is a better choice because the it make the agent more flexible.

Our needs might be different from kata (e.g., image paths are shared on the host). One of the questions is do we need unique container ids paths independent of what the image name is? Can the image pull response `image_ref` used as the host path identifier instead of `container_id`? /cc @haosanzi 

We seem to have the container ID generation logic implemented in two places (shim and agent). With the kata approach our shim could drop the ID generation part but the current image-rs occlum implementation expects someone has created the sefs paths... and given that constraint the logic in agent-enclave could be to just check valid and non-empty `cid`s.

Given this it sounds to me this PR can change to just validate the `cid` is provided and it's valid. And then the dev tool needs fixing... I'm fine with the kata approach too but then we need to think how the sefs paths are created.

Reply:
@haosanzi thanks, I also came to the same conclusion about the timing.

> Given this it sounds to me this PR can change to just validate the `cid` is provided and it's valid. And then the dev tool needs fixing...

do you agree this is the right approach then?

Reply:
yes, I agree with you. Thank you very mych

Reply:
Codes have be updated.